omnigibson/utils/usd_utils.py

```diff
@@ -203,78 +235,89 @@ def initialize_view(cls):
         # Note that omni's ordering is based on the top-down object ordering path on the USD stage, which coincidentally
         # matches the same ordering we store objects in our registry. So the mapping we generate from our registry
         # mapping aligns with omni's ordering!
-        i = 0
-        cls._PATH_TO_COL_IDX = dict()
-        for obj in og.sim.scene.objects:
-            if obj.prim_type == PrimType.RIGID:
-                for link in obj.links.values():
-                    if not link.kinematic_only:
-                        cls._PATH_TO_COL_IDX[link.prim_path] = i
-                        i += 1
+        column_filters = self.get_column_filters()
+        for scene_idx, filters in column_filters.items():
+            self._PATH_TO_COL_IDX[scene_idx] = dict()
+            for i, link_path in enumerate(filters):
+                self._PATH_TO_COL_IDX[scene_idx][link_path] = i
+                self._PATH_TO_SCENE_IDX[link_path] = scene_idx
 
         # If there are no valid objects, clear the view and terminate early
-        if i == 0:
-            cls._CONTACT_VIEW = None
+        if len(column_filters) == 0:
+            self._CONTACT_VIEW = dict()
             return
 
+        # Get the row filters too
+        row_filters = self.get_row_filters()
+
         # Generate rigid body view, making sure to update the simulation first (without physics) so that the physx
         # backend is synchronized with any newly added objects
         # We also suppress the omni tensor plugin from giving warnings we expect
         og.sim.pi.update_simulation(elapsedStep=0, currentTime=og.sim.current_time)
         with suppress_omni_log(channels=["omni.physx.tensors.plugin"]):
-            cls._CONTACT_VIEW = og.sim.physics_sim_view.create_rigid_contact_view(
-                pattern="/World/*/*",
-                filter_patterns=list(cls._PATH_TO_COL_IDX.keys()),
-            )
+            for scene_idx, _ in enumerate(og.sim.scenes):
+                # TODO: How to make this work with the floor plane?
+                # If there are no collidable objects in the scene, skip.
+                if len(column_filters[scene_idx]) == 0:
+                    continue
+
+                self._CONTACT_VIEW[scene_idx] = og.sim.physics_sim_view.create_rigid_contact_view(
```

Comment:
is it possible to create a single view across all scenes? or does this not work and / or not provide us any perf benefit? @cgokmen

Reply:
You end up with a huge contact matrix that is block-diagonal e.g. it's 99% zeros because objects in diff scenes cannot come into contact. Wastes a bunch of space and compute to work with it.